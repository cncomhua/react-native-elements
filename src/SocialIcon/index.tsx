import { withTheme } from '../config';
import { SocialIcon, SocialIconProps, SocialMediaType } from './SocialIcon';

export { SocialIcon };
export type { SocialIconProps, SocialMediaType };
export default withTheme(SocialIcon, 'SocialIcon');
