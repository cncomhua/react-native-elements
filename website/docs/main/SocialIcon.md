---


id: socialicon


title: SocialIcon


slug: /socialicon

---



import Usage from './usage/SocialIcon/SocialIcon.md'



SocialIcons are visual cues to online and social media networks. We offer a varied range of social icons.



## Usage


<Usage />

---


## Props

### SocialIcon




## Reference

### SocialIcon

