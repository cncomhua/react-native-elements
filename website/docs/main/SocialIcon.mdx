---
id: socialicon

title: SocialIcon

slug: /socialicon
---

## import Usage from "./usage/SocialIcon/SocialIcon.mdx";

import Props from './props/SocialIcon.mdx';

import { SocialIcon } from 'react-native-elements';

import Tabs from '@theme/Tabs';

import TabItem from '@theme/TabItem';

SocialIcons are visual cues to online and social media networks. We offer a varied range of social icons.

## Usage

<Usage />

## Props

<Props />
