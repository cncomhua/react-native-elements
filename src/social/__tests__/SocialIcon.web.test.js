import React from 'react';
import {
  Text,
  TouchableHighlight,
  TouchableWithoutFeedback,
  View,
} from 'react-native';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import parseSamples, {
  assignRef,
  genRefId,
} from 'enzyme-styleguidist-sample-parser';
import options, {
  snapShot,
  buildJsxForGuideMethod,
  ensureCalled,
  onlyEnsureCalled,
  onlySnapshots,
} from '../../../samples';
import SocialIcon from '../SocialIcon';

const props = {
  noProps: {
    component: SocialIcon,
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  title: {
    component: SocialIcon,
    props: { type: 'medium', title: 'medium', button: true },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
  },
  type: {
    component: SocialIcon,
    props: { type: 'medium' },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
  },
  raised: {
    component: SocialIcon,
    props: { type: 'medium', raised: false },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
  },
  button: {
    component: SocialIcon,
    props: { type: 'medium', button: true },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
  },
  onPress: {
    component: SocialIcon,
    props: {
      type: 'medium',
      onPress: () => {
        console.log('pressed');
      },
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
  },
  onLongPress: {
    component: SocialIcon,
    props: {
      type: 'medium',
      onLongPress: () => {
        console.log('pressed, long');
      },
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
  },
  light: {
    component: SocialIcon,
    props: { type: 'medium', light: true },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  iconStyle: {
    component: SocialIcon,
    props: {
      type: 'medium',
      button: true,
      iconStyle: { transform: [{ rotateZ: '-45deg' }] },
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
  },
  style: {
    component: SocialIcon,
    props: {
      type: 'medium',
      button: true,
      style: { width: 120, transform: [{ rotateZ: '-45deg' }] },
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
  },
  iconColor: {
    component: SocialIcon,
    props: { type: 'medium', iconColor: '#071' },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
  },
  underlayColor: {
    component: SocialIcon,
    props: {
      type: 'medium',
      onPress: () => {},
      underlayColor: 'yellow',
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
  },
  iconSize: {
    component: SocialIcon,
    props: { type: 'medium', iconSize: 40 },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
  },
  component: {
    component: SocialIcon,
    props: { type: 'medium', component: TouchableWithoutFeedback },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {
      getJsxString: () => {
        return `const TouchableWithoutFeedback = RN.TouchableWithoutFeedback;
  <SocialIcon
    type='medium'
    component={TouchableWithoutFeedback}
  />`;
      },
    },
  },
  fontFamily: {
    component: SocialIcon,
    props: {
      type: 'medium',
      fontFamily: 'Courier New',
      title: 'medium',
      button: true,
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
  },
  fontWeight: {
    component: SocialIcon,
    props: {
      type: 'medium',
      fontWeight: 'bold',
      title: 'medium',
      button: true,
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
  },
  fontStyle: {
    component: SocialIcon,
    props: {
      type: 'medium',
      fontStyle: { padding: 50 },
      title: 'medium',
      button: true,
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
  },
  disabled: {
    component: SocialIcon,
    props: { type: 'medium', disabled: true },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
  },
  loading: {
    component: SocialIcon,
    props: { type: 'medium', loading: true, title: 'medium', button: true },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
  },
};

const samples = {
  props,
};

parseSamples(
  {
    Icons: { sectionComponents: { SocialIcon: { samples } } },
  },
  options
);
describe('SocialIcon component', () => {
  it('should render without issues', () => {
    const component = shallow(<SocialIcon type="twitter" />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should show loading indicator', () => {
    const component = shallow(<SocialIcon type="twitter" loading />);

    expect(component.find('ActivityIndicator').length).toBe(1);
  });

  it('should render light social icon', () => {
    const component = shallow(
      <SocialIcon light raised={false} type="medium" />
    );

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render social icon button', () => {
    const component = shallow(
      <SocialIcon title="Sign In With Facebook" button type="facebook" />
    );

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should have onPress event', () => {
    const onPress = jest.fn();
    const component = shallow(<SocialIcon onPress={onPress} type="gitlab" />);

    component.simulate('press');
    expect(onPress).toHaveBeenCalled();
  });
});
