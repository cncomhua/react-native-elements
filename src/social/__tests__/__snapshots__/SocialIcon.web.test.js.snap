// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icons: SocialIcon: Props Icons: SocialIcon: [ prop: button ] enzyme-shallow: snapshot: Icons: SocialIcon: [ prop: button ] enzyme-shallow: snapshot 1`] = `
<View
  disabled={false}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#02b875",
      "borderRadius": 30,
      "flexDirection": "row",
      "justifyContent": "center",
      "margin": 7,
      "paddingBottom": 14,
      "paddingTop": 14,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Icon
      color="white"
      name="medium"
      size={24}
      style={
        Array [
          undefined,
        ]
      }
    />
  </View>
</View>
`;

exports[`Icons: SocialIcon: Props Icons: SocialIcon: [ prop: component ] enzyme-shallow: snapshot: Icons: SocialIcon: [ prop: component ] enzyme-shallow: snapshot 1`] = `
<View
  disabled={false}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#02b875",
      "borderRadius": 30,
      "flexDirection": "row",
      "height": 52,
      "justifyContent": "center",
      "margin": 7,
      "width": 52,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Icon
      color="white"
      name="medium"
      size={24}
      style={
        Array [
          undefined,
        ]
      }
    />
  </View>
</View>
`;

exports[`Icons: SocialIcon: Props Icons: SocialIcon: [ prop: disabled ] enzyme-shallow: snapshot: Icons: SocialIcon: [ prop: disabled ] enzyme-shallow: snapshot 1`] = `
<View
  disabled={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#02b875",
      "borderRadius": 30,
      "flexDirection": "row",
      "height": 52,
      "justifyContent": "center",
      "margin": 7,
      "width": 52,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Icon
      color="white"
      name="medium"
      size={24}
      style={
        Array [
          undefined,
        ]
      }
    />
  </View>
</View>
`;

exports[`Icons: SocialIcon: Props Icons: SocialIcon: [ prop: fontFamily ] enzyme-shallow: snapshot: Icons: SocialIcon: [ prop: fontFamily ] enzyme-shallow: snapshot 1`] = `
<View
  disabled={false}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#02b875",
      "borderRadius": 30,
      "flexDirection": "row",
      "justifyContent": "center",
      "margin": 7,
      "paddingBottom": 14,
      "paddingTop": 14,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Icon
      color="white"
      name="medium"
      size={24}
      style={
        Array [
          undefined,
        ]
      }
    />
    <TextElement
      style={
        Object {
          "color": "white",
          "fontFamily": "Courier New",
          "marginLeft": 15,
        }
      }
    >
      medium
    </TextElement>
  </View>
</View>
`;

exports[`Icons: SocialIcon: Props Icons: SocialIcon: [ prop: fontStyle ] enzyme-shallow: snapshot: Icons: SocialIcon: [ prop: fontStyle ] enzyme-shallow: snapshot 1`] = `
<View
  disabled={false}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#02b875",
      "borderRadius": 30,
      "flexDirection": "row",
      "justifyContent": "center",
      "margin": 7,
      "paddingBottom": 14,
      "paddingTop": 14,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Icon
      color="white"
      name="medium"
      size={24}
      style={
        Array [
          undefined,
        ]
      }
    />
    <TextElement
      style={
        Object {
          "color": "white",
          "marginLeft": 15,
          "padding": 50,
        }
      }
    >
      medium
    </TextElement>
  </View>
</View>
`;

exports[`Icons: SocialIcon: Props Icons: SocialIcon: [ prop: fontWeight ] enzyme-shallow: snapshot: Icons: SocialIcon: [ prop: fontWeight ] enzyme-shallow: snapshot 1`] = `
<View
  disabled={false}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#02b875",
      "borderRadius": 30,
      "flexDirection": "row",
      "justifyContent": "center",
      "margin": 7,
      "paddingBottom": 14,
      "paddingTop": 14,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Icon
      color="white"
      name="medium"
      size={24}
      style={
        Array [
          undefined,
        ]
      }
    />
    <TextElement
      style={
        Object {
          "color": "white",
          "fontWeight": "bold",
          "marginLeft": 15,
        }
      }
    >
      medium
    </TextElement>
  </View>
</View>
`;

exports[`Icons: SocialIcon: Props Icons: SocialIcon: [ prop: iconColor ] enzyme-shallow: snapshot: Icons: SocialIcon: [ prop: iconColor ] enzyme-shallow: snapshot 1`] = `
<View
  disabled={false}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#02b875",
      "borderRadius": 30,
      "flexDirection": "row",
      "height": 52,
      "justifyContent": "center",
      "margin": 7,
      "width": 52,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Icon
      color="#071"
      name="medium"
      size={24}
      style={
        Array [
          undefined,
        ]
      }
    />
  </View>
</View>
`;

exports[`Icons: SocialIcon: Props Icons: SocialIcon: [ prop: iconSize ] enzyme-shallow: snapshot: Icons: SocialIcon: [ prop: iconSize ] enzyme-shallow: snapshot 1`] = `
<View
  disabled={false}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#02b875",
      "borderRadius": 30,
      "flexDirection": "row",
      "height": 52,
      "justifyContent": "center",
      "margin": 7,
      "width": 52,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Icon
      color="white"
      name="medium"
      size={40}
      style={
        Array [
          undefined,
        ]
      }
    />
  </View>
</View>
`;

exports[`Icons: SocialIcon: Props Icons: SocialIcon: [ prop: iconStyle ] enzyme-shallow: snapshot: Icons: SocialIcon: [ prop: iconStyle ] enzyme-shallow: snapshot 1`] = `
<View
  disabled={false}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#02b875",
      "borderRadius": 30,
      "flexDirection": "row",
      "justifyContent": "center",
      "margin": 7,
      "paddingBottom": 14,
      "paddingTop": 14,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Icon
      color="white"
      name="medium"
      size={24}
      style={
        Object {
          "transform": Array [
            Object {
              "rotateZ": "-45deg",
            },
          ],
        }
      }
    />
  </View>
</View>
`;

exports[`Icons: SocialIcon: Props Icons: SocialIcon: [ prop: light ] enzyme-shallow: snapshot: Icons: SocialIcon: [ prop: light ] enzyme-shallow: snapshot 1`] = `
<View
  disabled={false}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "borderRadius": 30,
      "flexDirection": "row",
      "height": 52,
      "justifyContent": "center",
      "margin": 7,
      "width": 52,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Icon
      color="#02b875"
      name="medium"
      size={24}
      style={
        Array [
          undefined,
        ]
      }
    />
  </View>
</View>
`;

exports[`Icons: SocialIcon: Props Icons: SocialIcon: [ prop: loading ] enzyme-shallow: snapshot: Icons: SocialIcon: [ prop: loading ] enzyme-shallow: snapshot 1`] = `
<View
  disabled={false}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#02b875",
      "borderRadius": 30,
      "flexDirection": "row",
      "justifyContent": "center",
      "margin": 7,
      "paddingBottom": 14,
      "paddingTop": 14,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Icon
      color="white"
      name="medium"
      size={24}
      style={
        Array [
          undefined,
        ]
      }
    />
    <TextElement
      style={
        Object {
          "color": "white",
          "marginLeft": 15,
        }
      }
    >
      medium
    </TextElement>
    <ActivityIndicator
      animating={true}
      color="white"
      hidesWhenStopped={true}
      size="large"
      style={
        Object {
          "height": 0,
          "marginHorizontal": 10,
        }
      }
    />
  </View>
</View>
`;

exports[`Icons: SocialIcon: Props Icons: SocialIcon: [ prop: noProps ] enzyme-shallow: snapshot: Icons: SocialIcon: [ prop: noProps ] enzyme-shallow: snapshot 1`] = `
<View
  disabled={false}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": undefined,
      "borderRadius": 30,
      "flexDirection": "row",
      "height": 52,
      "justifyContent": "center",
      "margin": 7,
      "width": 52,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Icon
      color="white"
      size={24}
      style={
        Array [
          undefined,
        ]
      }
    />
  </View>
</View>
`;

exports[`Icons: SocialIcon: Props Icons: SocialIcon: [ prop: onLongPress ] enzyme-shallow: snapshot: Icons: SocialIcon: [ prop: onLongPress ] enzyme-shallow: snapshot 1`] = `
<TouchableHighlight
  activeOpacity={0.85}
  disabled={false}
  onLongPress={[Function]}
  onPress={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#02b875",
      "borderRadius": 30,
      "flexDirection": "row",
      "height": 52,
      "justifyContent": "center",
      "margin": 7,
      "width": 52,
    }
  }
  underlayColor="#02b875"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Icon
      color="white"
      name="medium"
      size={24}
      style={
        Array [
          undefined,
        ]
      }
    />
  </View>
</TouchableHighlight>
`;

exports[`Icons: SocialIcon: Props Icons: SocialIcon: [ prop: onPress ] enzyme-shallow: snapshot: Icons: SocialIcon: [ prop: onPress ] enzyme-shallow: snapshot 1`] = `
<TouchableHighlight
  activeOpacity={0.85}
  disabled={false}
  onLongPress={[Function]}
  onPress={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#02b875",
      "borderRadius": 30,
      "flexDirection": "row",
      "height": 52,
      "justifyContent": "center",
      "margin": 7,
      "width": 52,
    }
  }
  underlayColor="#02b875"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Icon
      color="white"
      name="medium"
      size={24}
      style={
        Array [
          undefined,
        ]
      }
    />
  </View>
</TouchableHighlight>
`;

exports[`Icons: SocialIcon: Props Icons: SocialIcon: [ prop: raised ] enzyme-shallow: snapshot: Icons: SocialIcon: [ prop: raised ] enzyme-shallow: snapshot 1`] = `
<View
  disabled={false}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#02b875",
      "borderRadius": 48,
      "flexDirection": "row",
      "height": 52,
      "justifyContent": "center",
      "margin": 7,
      "width": 52,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Icon
      color="white"
      name="medium"
      size={24}
      style={
        Array [
          undefined,
        ]
      }
    />
  </View>
</View>
`;

exports[`Icons: SocialIcon: Props Icons: SocialIcon: [ prop: style ] enzyme-shallow: snapshot: Icons: SocialIcon: [ prop: style ] enzyme-shallow: snapshot 1`] = `
<View
  disabled={false}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#02b875",
      "borderRadius": 30,
      "flexDirection": "row",
      "justifyContent": "center",
      "margin": 7,
      "paddingBottom": 14,
      "paddingTop": 14,
      "transform": Array [
        Object {
          "rotateZ": "-45deg",
        },
      ],
      "width": 120,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Icon
      color="white"
      name="medium"
      size={24}
      style={
        Array [
          undefined,
        ]
      }
    />
  </View>
</View>
`;

exports[`Icons: SocialIcon: Props Icons: SocialIcon: [ prop: title ] enzyme-shallow: snapshot: Icons: SocialIcon: [ prop: title ] enzyme-shallow: snapshot 1`] = `
<View
  disabled={false}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#02b875",
      "borderRadius": 30,
      "flexDirection": "row",
      "justifyContent": "center",
      "margin": 7,
      "paddingBottom": 14,
      "paddingTop": 14,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Icon
      color="white"
      name="medium"
      size={24}
      style={
        Array [
          undefined,
        ]
      }
    />
    <TextElement
      style={
        Object {
          "color": "white",
          "marginLeft": 15,
        }
      }
    >
      medium
    </TextElement>
  </View>
</View>
`;

exports[`Icons: SocialIcon: Props Icons: SocialIcon: [ prop: type ] enzyme-shallow: snapshot: Icons: SocialIcon: [ prop: type ] enzyme-shallow: snapshot 1`] = `
<View
  disabled={false}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#02b875",
      "borderRadius": 30,
      "flexDirection": "row",
      "height": 52,
      "justifyContent": "center",
      "margin": 7,
      "width": 52,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Icon
      color="white"
      name="medium"
      size={24}
      style={
        Array [
          undefined,
        ]
      }
    />
  </View>
</View>
`;

exports[`Icons: SocialIcon: Props Icons: SocialIcon: [ prop: underlayColor ] enzyme-shallow: snapshot: Icons: SocialIcon: [ prop: underlayColor ] enzyme-shallow: snapshot 1`] = `
<TouchableHighlight
  activeOpacity={0.85}
  disabled={false}
  onLongPress={[Function]}
  onPress={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#02b875",
      "borderRadius": 30,
      "flexDirection": "row",
      "height": 52,
      "justifyContent": "center",
      "margin": 7,
      "width": 52,
    }
  }
  underlayColor="yellow"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Icon
      color="white"
      name="medium"
      size={24}
      style={
        Array [
          undefined,
        ]
      }
    />
  </View>
</TouchableHighlight>
`;

exports[`SocialIcon component should render light social icon 1`] = `
<View
  disabled={false}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "borderRadius": 30,
      "flexDirection": "row",
      "justifyContent": "center",
      "margin": 7,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Icon
      color="#02b875"
      name="medium"
      size={24}
      style={
        Array [
          undefined,
        ]
      }
    />
  </View>
</View>
`;

exports[`SocialIcon component should render social icon button 1`] = `
<View
  disabled={false}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#3b5998",
      "borderRadius": 30,
      "flexDirection": "row",
      "justifyContent": "center",
      "margin": 7,
      "paddingBottom": 14,
      "paddingTop": 14,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Icon
      color="white"
      name="facebook"
      size={24}
      style={
        Array [
          undefined,
        ]
      }
    />
    <TextElement
      style={
        Object {
          "color": "white",
          "marginLeft": 15,
        }
      }
    >
      Sign In With Facebook
    </TextElement>
  </View>
</View>
`;

exports[`SocialIcon component should render without issues 1`] = `
<View
  disabled={false}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#00aced",
      "borderRadius": 30,
      "flexDirection": "row",
      "height": 52,
      "justifyContent": "center",
      "margin": 7,
      "width": 52,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Icon
      color="white"
      name="twitter"
      size={24}
      style={
        Array [
          undefined,
        ]
      }
    />
  </View>
</View>
`;
